### PR TITLE
Use full datapath name when querying datapaths with ovs-appctl

### PR DESCRIPTION
**Pull request by JM1**
When OVN-Kubernetes lists all datapaths with `ovs-appctl dpctl/dump-dps`, it stores both a datapath's name and type. But it then continues to use only each datapath's name when querying `ovs-appctl dpctl/show`.

For system datapaths this works because `ovs-appctl dpctl/show` will assume the `system` type if no type has been set. However, this breaks when using other datapath types such as `netdev`, causing `ovs-appctl` to fail with:

```
ovs-vswitchd: opening datapath ovs-netdev failed (No such device)
```

This change keeps using a datapath's name as an identifier for metrics but includes the type when querying ovs-appctl.

Whether a datapath's name without a type is sufficient for identifying datapaths uniquely in OpenShift and Open vSwitch is left for future work.

### CODE REVIEW

**Review by tssurya — COMMENTED** [review]
LGTM overall

**Comment by JM1** [comment]
@jcaamano @tssurya Thank you for reviewing ☺️ Hopefully the latest revision is better at explaining the WHY?!

File: go-controller/pkg/metrics/ovs.go [hunk 1]
```diff
@@ -366,27 +366,29 @@ func getOvsDatapaths(ovsAppctl ovsClient) (datapathsList []string, err error) {
...
 			err = fmt.Errorf("recovering from a panic while parsing the ovs-appctl dpctl/"+
-				"show %s output : %v", datapathName, r)
+				"show %s output : %v", datapath, r)
 		}
 	}()
 
-	for _, datapathName = range datapaths {
-		stdout, stderr, err = ovsAppctl("dpctl/show", datapathName)
+	for _, datapath = range datapaths {
+		datapathName := strings.Split(datapath, "@")[1]
```
[the remark this answers is not in the document]
Comment (JM1) [reply go-controller/pkg/metrics/ovs.go]: True. Latest patch moves datapathName down, closer to its usage. I also added comments to explain the name/type split.

File: go-controller/pkg/metrics/ovs.go [hunk 2]
```diff
@@ -366,27 +366,29 @@ func getOvsDatapaths(ovsAppctl ovsClient) (datapathsList []string, err error) {
 			return nil, fmt.Errorf("datapath %s is not of format Type@Name", output)
 		}
 		metricOvsDp.WithLabelValues(datapathName, datapathType).Set(1)
-		datapathsList = append(datapathsList, datapathName)
+		datapathsList = append(datapathsList, output)
```
[the remark this answers is not in the document]
Comment (JM1) [reply go-controller/pkg/metrics/ovs.go:369]: Added your self-note as a comment 😉

**Review by jcaamano — APPROVED** [review]

**Comment by jcaamano** [comment]
known flake: https://github.com/ovn-org/ovn-kubernetes/issues/4060

**Comment by JM1** [comment]
@jcaamano Thank you 🥳

**Pull request merged**